### ISSUE TITLE
Confusion in Pretrained arguement

### ISSUE THREAD

**Issue by Areebkhan02** [ultralytics/ultralytics#15013]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I'm confused about a few things while training models using Ultralytics YOLOv8:
    1. Pre-train Argument in model.train
        ◦ If I provide YOLO with a .pt file and set the pretrain argument to True, will it use the COCO weights or the weights from my .pt file?
        ◦ If I set pretrain to False, will the model train from scratch with weights being randomly initialized?
    2. Continuing Training from a Saved Model
        ◦ If I train a model for 100 epochs and then use the last.pt file to train the model for another 100 epochs, is it accurate to say that the model has been trained for a total of 200 epochs, or is there something else at play?
    3. Training from Scratch with a .yaml File
        ◦ If I want to train the model from scratch and provide a .yaml file, is it necessary to set the pretrain argument to False, or is this not a concern?

**Comment by pderrenger** [comment]
@Areebkhan02 thank you for your questions. Let's clarify each point:

1. **Pre-trained Argument in `model.train`**:
   - If you provide a `.pt` file and set `pretrained=True`, the model will use the weights from your `.pt` file, not the COCO weights.
   - If `pretrained=False`, the model will train from scratch with randomly initialized weights.

2. **Continuing Training from a Saved Model**:
   - Yes, if you train a model for 100 epochs and then continue training from `last.pt` for another 100 epochs, it effectively means the model has been trained for a total of 200 epochs.

3. **Training from Scratch with a `.yaml` File**:
   - If you want to train from scratch using a `.yaml` file, you should set `pretrained=False` to ensure the model starts with randomly initialized weights.

Feel free to reach out if you have any more questions!

**Comment by Y-T-G** [comment]
For detection model, `pretrained` argument only works if you provide a path to a checkpoint to the argument. It will then load the weights from that checkpoint to the original model passed using the `model` argument. `pretrained=True` or `pretrained=False` doesn't do anything for detection models. The boolean value only works for classification model.

**Comment by Areebkhan02** [comment]
@Y-T-G  so you mean to say for the detection task if I write 

`model = YOLO('yolov8n.pt') 
result = model.train(data="data/path", pretrained = False) 
`
it will still load .pt file weights ?

**Comment by Y-T-G** [comment]
Yes. If you want to avoid loading weights, you have to use `yolov8n.yaml`.

**Comment by Invictu520** [comment]
@pderrenger Is it necessary to set pretrained to false? 
Because I am a bit confused.  On the ultralytics website it says nothing about it. 
It just says that if I want to build a model from scratch this is what I have to write:
`model = YOLO("yolov8n.yaml")  # build a new model from YAML`

So I did train a model using this code but in the args file after the training it says
pretrained=true.

So if initializing the model with the .yaml file is not enough why is that not mentioned anywhere?

**Comment by Y-T-G** [comment]
@Invictu520 It does not have an effect. As long as you used yaml, it was trained from scratch.
